### DISCUSSION TITLE
How to use with Kubernetes GKE traefik module or Java server

### DISCUSSION THREAD

**Discussion by Xa3lis** [posva/unplugin-vue-router#416] [Questions and Help]
Hello ! 

We have a problem in our project where the `[` and the `]` are blocked by the GKE traefik module (Unauthorized characters i guess)

We were using [Vue plugin pages](https://github.com/hannoeru/vite-plugin-pages) before and it had a "nuxt routeStyle" (https://github.com/hannoeru/vite-plugin-pages#routestyle) and it was using a `pages/xxx/_dynamicParam/index.vue` format

The `_xxx` was working, but the `[xxx]` format is blocked

Is there a way to use `_xxx` format within this lib or is there a way to URL encode the path before it's handled by Vite ?

Thank's !

**Comment by posva** [comment]
I couldn't find a lot of information about _GKE traefik module_ other than it being related to Kubernetes. If the characters are unauthorized, it's probably necessary to configure it but I don't know how so I would move this to a discussion.

Regarding the old format `_xxx`, there is no way to use it and no plans to support it since Nuxt 3 doesn't support it either, it creates fragmentation in the ecosystem for tools, and it would require a **considerable rewrite** to this plugin and any existing extensions

**Comment by Xa3lis** [comment]
After some investigation, the problem might be linked to the java server serving our front app, and the `[` / `]` are not allowed in url java side (RFC3986)

Is there a way to "trick" our local dev vite server to use URL Encode when doing import ?

We tried with curl to use `src/pages/deliveries/%5Bid%5D/index.vue` and it's perfectly working
